<launch>
  <arg name="planning" default="ompl"/>
  <!-- By default, we are not in debug mode -->
  <arg name="debug" default="false"/>
  <arg name="load_robot_description" default="true"/>
  <arg name="gui" default="true"/>
  <arg name="kinematics_template" default="$(find sr_moveit_hand_config)/config/kinematics_template.yaml"/>
  
  <!-- Broadcast static tf for robot root -->
  <node name="virtual_joint_broadcaster" pkg="sr_moveit_hand_config" type="virtual_joint_broadcaster.py" output="screen" />
  
  <include file="$(find sr_moveit_hand_config)/launch/planning_context.launch">
    <arg name="kinematics_template" value="kinematics_template" />
  </include>
  
  <node pkg="sr_utilities_common" type="timed_roslaunch.sh" args="2 sr_moveit_hand_config move_group.launch 
    allow_trajectory_execution:=true 
    fake_execution:=false 
    info:=true 
    debug:=$(arg debug) 
    publish_monitored_planning_scene:=true
    pipeline:=$(arg planning)
    kinematics_template:=$(arg kinematics_template)" name="timed_roslaunch" output="screen"/>

  <!-- Run Rviz and load the default config to see the state of the move_group node -->
  <group if="$(arg gui)">
    <node pkg="sr_utilities_common" type="timed_roslaunch.sh" args="2 sr_moveit_hand_config moveit_rviz.launch 
      config:=true 
      debug:=$(arg debug)
      kinematics_template:=($arg kinematics_template)" name="timed_roslaunch_rviz" output="screen"
    />
  </group>
</launch>
